AWS/AWS_Setup.sh

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
+
+#This script is intended to automatically configure new EC2 instances on launch. To make use of it; copy the contents to your clipboard, log-in to the AWS console, select EC2, and click on 
+#the "launch template" option in the left-hand navigation pane. Make sure you are using the Ubuntu 18.04 AMI (with a t2.micro instance if you want to stay within the bounds of free tier), and scroll down
+#to advanced details. Click the dropdown, and paste the contents of this file you previously copied into user data, and click "create template version." You will now be able to launch new instances with
+#this template via the launch instances drop down on the EC2 dashboard.
+IMAGE="<image name>"
+#Install Docker
+(
+set -e
+curl -fsSL https://get.docker.com -o get-docker.sh
+sh get-docker.sh
+docker pull "$IMAGE"
+set +e
+)
+
+#Create .env, edit as necessary
+mkdir /home/ubuntu/docker_config && cat > /home/ubuntu/docker_config/.env <<EOF
+LOCAL_DATA=/app/data
+DATASTORE_NAME=S3
+DATASTORE_PATH=performance-tracker-test
+TMP_DIR=/app/data/tmp
+METRO_LINES=801,802,803,804,805,806
+METRO_AGENCY=lametro-rail
+TIMEZONE=America/Los_Angeles
+SCHEDULE_URL=https://gitlab.com/LACMTA/gtfs_rail/raw/master/gtfs_rail.zip
+VEHICLE_API_URL=http://webservices.nextbus.com/service/publicJSONFeed
+LOG_TIMESTAMPS=TRUE
+AWS_ACCESS_KEY_ID=<access key id>
+AWS_SECRET_ACCESS_KEY=<secret key>
+EOF
+
+#allow for timestamps on STDERR
+
+(
+set -e
+mkdir /home/ubuntu/utils
+
+cat > /home/ubuntu/utils/timestamp_errors.sh <<EOF
+#!/bin/bash
+while read -r line; do
+   echo "[\$(date +%Y-%m-%d\ %H:%M:%S)] \$line" >> /home/ubuntu/logs/errors.log
+done
+EOF
+chmod 755 /home/ubuntu/utils/timestamp_errors.sh
+chown -R ubuntu /home/ubuntu/utils
+set +e
+)
+
+#Create Shell Scripts
+mkdir /home/ubuntu/logs
+touch /home/ubuntu/logs/errors.log
+touch /home/ubuntu/logs/output.log
+chown -R ubuntu /home/ubuntu/logs 
+
+(
+set -e
+cat > /home/ubuntu/daily_at_2.sh <<EOF
+#!/bin/bash
+sudo docker run --rm --env-file /home/ubuntu/docker_config/.env -v /home/ubuntu/data/tmp:/app/data/tmp "$IMAGE" bash ./tasks/DAILY_AT_2AM.sh
```

Comment:
If we are using S3 as the "persistence" layer, then the volume mount shouldn't be necessary at all. On the other hand it will be necessary if using the filesystem.